tensorboard/webapp/app_routing/dirty_updates_registry_module.ts

```diff
@@ -14,63 +14,66 @@ limitations under the License.
 ==============================================================================*/
 import {Inject, ModuleWithProviders, NgModule, Optional} from '@angular/core';
 
-import {EXPS_WITH_DIRTY_UPDATES_TOKEN} from './dirty_updates_registry_types';
-import {DirtyUpdates} from './types';
+import {
+  DIRTY_UPDATES_TOKEN,
+  DirtyUpdatesSelector,
+} from './dirty_updates_registry_types';
 
 @NgModule()
-export class DirtyUpdatesRegistryModule {
-  private readonly dirtyUpdates: DirtyUpdates | null = null;
+export class DirtyUpdatesRegistryModule<State, Updates> {
+  private readonly dirtyUpdatesSelectors: DirtyUpdatesSelector<
+    State,
+    Updates
+  >[] = [];
 
   constructor(
-    @Optional() @Inject(EXPS_WITH_DIRTY_UPDATES_TOKEN) eidsList: string[][]
+    @Optional()
+    @Inject(DIRTY_UPDATES_TOKEN)
+    dirtyUpdatesSelectorFactories: Array<
+      () => DirtyUpdatesSelector<State, Updates>
+    > | null
```

Comment:
Do note that Angular dependency injector is the one that would be invoking the factory and inject only `DirtyUpdatesSelector<State, Updates>`.  That said, I am surprised the test is passing.

Reply:
Fixed. I think the test was problematic as well and is now fixed.